### ISSUE TITLE
Quest: POI without name - example with shops

### ISSUE THREAD

**Issue by Jibec** [westnordost/StreetComplete#309]
hi,

can we please have places without names? I think many types of POI could benefit such quests, but depending on the type of POI, the name can be in a relation (group of buildings & stuff).

Here is an "easy" example with [shops](http://wiki.openstreetmap.org/wiki/Key:shop):
```
[out:json][timeout:25];
(
  node["shop"][!"name"]({{bbox}});
  way["shop"][!"name"]({{bbox}});
);
out skel;
```

I assume it should also be the same for [amenities](http://wiki.openstreetmap.org/wiki/Key:amenity), but it's a more difficult request that should be discussed. But if most pharmacy have a name (at least in France), a "bicycle_parking" won't.

thanks for this great application, you brought me back to cartography :)

**Comment by matkoniecz** [comment]
Many shops have no names in Poland. Such quest would further encourage to add descriptions in name field ("name=shoe store" type tagging is quite popular).

note

> only those quests are created which contain very few false positives.

in readme. Is there some region where almost all shops are named?

**Comment by westnordost** [comment]
It's implemented already but disabled. It's no use to ask the user for a random location on the map what name it has. It has to be named somehow but there is no dictionary from tag to translation currently.

Am 11. Juni 2017 11:07:44 MESZ schrieb Jean-Baptiste <notifications@github.com>:
>hi,
>
>can we please have places without names? I think many types of POI
>could benefit such quests, but depending on the type of POI, the name
>can be in a relation (group of buildings & stuff).
>
>Here is an "easy" example with
>[shops](http://wiki.openstreetmap.org/wiki/Key:shop):
>```
>[out:json][timeout:25];
>(
>  node["shop"][!"name"]({{bbox}});
>  way["shop"][!"name"]({{bbox}});
>);
>out skel;
>```
>
>I assume it should also be the same for
>[amenities](http://wiki.openstreetmap.org/wiki/Key:amenity), but it's a
>more difficult request that should be discussed. But if most pharmacy
>have a name (at least in France), a "bicycle_parking" won't.
>
>thanks for this great application, you brought me back to cartography
>:)
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/westnordost/StreetComplete/issues/309

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by Jibec** [comment]
@matkoniecz, indeed, I don't know how the naming convention if in each countries, I think it makes sense in France, even if some are just have a generic name or network brand, most have a real name.

**Comment by Jotam** [comment]
In Germany, virtually every shop has a name. So the feature would be very useful here. In many African countries, however, you will find hundreds of thousands of generic shops with no name at all. 
Therefore, I would love this kind of quest but would definitely want an option to switch it on/off in the settings. (This would be great with other quests as well, by the way.)

**Comment by dbdean** [comment]
We might be able to make use of the iD editors translations to solve the block on this issue.

the iD project already has a lot of tag translations in place, but I'm not sure how easy it might be too get it out.

For example, all the text for their presets is defined in https://github.com/openstreetmap/iD/blob/master/data/presets.yaml and translations are available.

**Comment by matkoniecz** [comment]
On topic of reusing collected data - https://github.com/osmlab/name-suggestion-index would be likely to be useful

**Comment by westnordost** [comment]
@matkoniecz true. I wonder if it were possible to have this per country.

@dbdean the iD presets are _relatively_ in a format that can be used by other applications. Still, I think, having a dictionary decoupled from other projects would be the best solution. Anyhow, the presets as is could not be used for this quest, because the quest asks i.e. "What is the name of this bakery?" or "What is the name of this supermarket?"
That works in English, but will not in other languages. In German:  "Was ist der Name **dieser** Bäckerei?" or "Was ist der Name **dieses** Supermarkt**es**"? Based on the genus of the word and the casus in which the word is used, the article and the word itself may be different.
So, in other words, such a dictionary would need to declare the words together with their genus (or whatever else serves to form the different casi) and it must be possible to get the words in these casi form the dictionary (so, additional country-wise logic required).

Workaround: Perhaps ask "Bakery: What is the name?" / "What is the name of this place (bakery)?". hmm, don't like it though :-/

**Comment by matkoniecz** [comment]
> I wonder if it were possible to have this per country.

Seems to be tracked by https://github.com/osmlab/name-suggestion-index/issues/11

> Workaround: Perhaps ask "Bakery: What is the name?" / "What is the name of this place (bakery)?". hmm, don't like it though :-/

"What is the name of object described as a bakery"? Still sounds clunky, but seems better than brackets.

Second possibility is to import English names and leave foreign labels for translations, but it would require a massive work form translators...

**Comment by dbdean** [comment]
I'm not familiar with how gender, case, (and singular/plural - does that come up too?) work with other, but would it make sense to produce a translation of 'What is the name of {this/these} {}?' for each of the possibilities (or at least the common ones), and then convince iD and/or osmlab to indicate that information in their databases so we can choose the right phrase in most cases

I think the [ICU](http://site.icu-project.org/) MessageFormat is designed for this sort of thing, and there appears to be an Android Implementation: https://developer.android.com/reference/android/icu/text/MessageFormat.html

Perfect is the enemy of good, and it would be useful to start collecting this sort of information, even if it might need a few internationalisation bugs ironed out on the way.

**Comment by dbdean** [comment]
Here are a few more links if anyone is interested, but it does seem like it is a different workflow than that that POEditor provides:

* https://format-message.github.io/icu-message-format-for-translators/index.html
* http://guigrpa.github.io/mady/
* https://github.com/messageformat/messageformat.js/issues/69
* https://github.com/messageformat/messageformat.js/issues/55

Can we ask the people doing the translation: "Please give us all the ways you can say 'What is the name of this {object}' in your language?' depending upon the gender, case and count of the object in question. Then we just need to assign those properties to each tag :)

Or we just ask for translations of every tag, and get it done, and make it better later?

**Comment by rugk** [comment]
Here some information on how such a pluralization thing can be designed: https://localization-guide.readthedocs.io/en/latest/l10n/pluralforms.html That solves this problem, at least…

**Comment by ENT8R** [comment]
I am currently trying to unblock this quest and I think it looks good so far...
I created some kind of multilingual dictionary just for OSM tags which can be found at https://github.com/ENT8R/tags2names This dictionary is used by the app to automatically retrieve the real name for a given OSM tag. The implementation is in an early state and I have already some ideas how to improve some things... But nevertheless, translating the dictionary will be a huge upcoming work for translators 😄 

Here are some screenshots:

**English:**

<img src="https://user-images.githubusercontent.com/25306497/34733801-f1f8f78e-f569-11e7-80d2-4fec35bbc1d0.png" width="300px"> <img src="https://user-images.githubusercontent.com/25306497/34733805-f44fb2d4-f569-11e7-8080-4a09cba52895.png" width="300px">

**And it even works in German 🎉**

<img src="https://user-images.githubusercontent.com/25306497/34733808-f76fbcb6-f569-11e7-87e6-13e6d5cacfcc.png" width="300px"> <img src="https://user-images.githubusercontent.com/25306497/34733814-f9da2522-f569-11e7-9a4a-cb1a9cb27785.png" width="300px">

What do you think about it?

**Comment by westnordost** [comment]
I answered you in your repository.

**Comment by matkoniecz** [comment]
direct link to the asnwer: https://github.com/ENT8R/tags2names/issues/1

**Comment by westnordost** [comment]
done!
